module/skill/QuinticWalk/src/QuinticWalk.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef MODULE_SKILL_QUINTICWALK_HPP
+#define MODULE_SKILL_QUINTICWALK_HPP
+
+#include <map>
+#include <memory>
+#include <nuclear>
+#include <vector>
+
+#include "extension/Behaviour.hpp"
+#include "extension/Configuration.hpp"
+
+#include "message/actuation/KinematicsModel.hpp"
+#include "message/behaviour/ServoCommand.hpp"
+#include "message/behaviour/state/Stability.hpp"
+
+#include "utility/input/ServoID.hpp"
+#include "utility/motion/WalkEngine.hpp"
+
+namespace module::skill {
+
+    class QuinticWalk : public ::extension::behaviour::BehaviourReactor {
+
+    public:
+        /// @brief Called by the powerplant to build and setup the QuinticWalk reactor.
+        explicit QuinticWalk(std::unique_ptr<NUClear::Environment> environment);
+
+        static constexpr int UPDATE_FREQUENCY = 200;
+
+    private:
+        // Reaction handle for the main update loop, disabling when not moving will save unnecessary CPU
+        // ReactionHandle update_handle{};
+        ReactionHandle imu_reaction{};
+
+        void calculate_joint_goals();
+        [[nodiscard]] float get_time_delta();
+        [[nodiscard]] std::unique_ptr<message::behaviour::ServoCommands> motion(
+            const std::vector<std::pair<utility::input::ServoID, float>>& joints);
+
+        struct Config {
+            Eigen::Vector3f max_step = Eigen::Vector3f::Zero();
+            float max_step_xy        = 0.0f;
+
+            bool imu_active           = true;
+            float imu_pitch_threshold = 0.0f;
+            float imu_roll_threshold  = 0.0f;
+
+            utility::motion::WalkingParameter params{};
+
+            std::map<utility::input::ServoID, float> jointGains{};
```

Comment:
```suggestion
            std::map<utility::input::ServoID, ServoState> jointGains{};
```
to match suggestion in previous comment